src/transformers/trainer_pt_utils.py

```diff
@@ -131,9 +131,10 @@ def nested_concat(tensors, new_tensors, padding_index=-100):
     Concat the `new_tensors` to `tensors` on the first dim and pad them on the second if needed. Works for tensors or
     nested list/tuples/dict of tensors.
     """
-    assert type(tensors) == type(
-        new_tensors
-    ), f"Expected `tensors` and `new_tensors` to have the same type but found {type(tensors)} and {type(new_tensors)}."
+    if not (isinstance(tensors, torch.Tensor) and isinstance(new_tensors, torch.Tensor)):
```

Comment:
We could simply test that both are instance of `torch.Tensor`. If you inherit from `torch.Tensor` is instance will be true

Reply:
@ArthurZucker Sorry I can't get your point. Could you let me know how can we simplify more?